generators/server/templates/dotnetcore/src/Project/Infrastructure/DatabaseStartup.cs.ejs

```diff
@@ -26,22 +37,55 @@ namespace <%= namespace %>.Infrastructure {
         public static IServiceCollection AddDatabaseModule(this IServiceCollection @this, IConfiguration configuration)
         {
             var entityFrameworkConfiguration = configuration.GetSection("EntityFramework");
+
+            <%_ switch(databaseType) {
+                case 'sqllite': _%>
             var connection = new SqliteConnection(new SqliteConnectionStringBuilder {
                 DataSource = entityFrameworkConfiguration["DataSource"]
             }.ToString());
 
             @this.AddDbContext<ApplicationDatabaseContext>(context => { context.UseSqlite(connection); });
+                <%_ break;
+                case 'mssql': _%>
+            var connection = configuration.GetConnectionString("AppDbContext");
+
+            @this.AddDbContext<ApplicationDatabaseContext>(options => options.UseSqlServer(connection));
+                <%_ break;
+                case 'postgres': _%>
+            var connection = configuration.GetConnectionString("AppDbContext");
+
+            @this.AddDbContext<ApplicationDatabaseContext>(context => { context.UseNpgsql(connection); });
+                <%_ break;
+                case 'mysql': _%>
+            var connection = configuration.GetConnectionString("AppDbContext");
+
+            @this.AddDbContext<ApplicationDatabaseContext>(options =>
+                options.UseMySql(connection, mySqlOptions =>
+                    mySqlOptions.ServerVersion(new ServerVersion(new Version(8, 0, 19), ServerType.MySql))
+            ));
+                <%_ break;
+                case 'oracle': _%>
+            var connection = configuration.GetConnectionString("AppDbContext");
+
+            @this.AddDbContext<ApplicationDatabaseContext>(options =>
+                options.UseOracle(connection));
+                <%_ break;
+            } _%>
 
             return @this;
         }
 
         public static IApplicationBuilder UseApplicationDatabase(this IApplicationBuilder @this,
             IServiceProvider serviceProvider, IHostEnvironment environment)
         {
-            if (environment.IsDevelopment() || environment.IsProduction()) {
-                var context = serviceProvider.GetRequiredService<ApplicationDatabaseContext>();
-                context.Database.OpenConnection();
-                context.Database.EnsureCreated();
+            if (environment.IsDevelopment() || environment.IsProduction())
```

Comment:
How many profiles can we have ?
dev/prod ? what else? the conditions looks weird.

Reply:
This is not our change, I can't tell you much about this piece of code. I just added scoping for this part, nothing more

Reply:
Ok don't touch it